pkg/util/proxyutil/proxyutil.go

```diff
@@ -130,3 +133,48 @@ func ApplyForwardIDHeader(req *http.Request, user identity.Requester) {
 		req.Header.Set(IDHeaderName, token)
 	}
 }
+
+func ApplyteamHTTPHeaders(req *http.Request, ds *datasources.DataSource, teams []int64) error {
+	teamHTTPHeadersJSON := datasources.TeamHTTPHeadersJSONData{}
+	if ds.JsonData != nil {
+		jsonData, err := ds.JsonData.MarshalJSON()
+		if err != nil {
+			return err
+		}
+		err = json.Unmarshal(jsonData, &teamHTTPHeadersJSON)
+		if err != nil {
+			return err
+		}
+
+		for teamID, headers := range teamHTTPHeadersJSON.TeamHTTPHeaders {
```

Comment:
If you were to move all of this logic into a method defined on the `TeamHTTPHeaders` struct, you would be able to share the functionality in both `ApplyTeamHTTPHeaders` and `TeamHTTPHeadersMiddleware.setHeaders`. This method would take a slice of team IDs and return the intersection of applicable headers.

Reply:
@cinaglia we decided we will merge this now as is and continue with refactoring and making this improved. we will add this as a refactoring step for later